compiler/rustc_middle/src/ty/flags.rs

```diff
@@ -161,6 +161,11 @@ impl FlagComputation {
                 self.add_substs(substs);
             }
 
+            &ty::TyAlias(_, substs) => {
+                self.add_flags(TypeFlags::HAS_TY_PROJECTION);
```

Comment:
This type flag is not really true, but we might rely on it in normalization code, for example. 

We might want to rename it to something that signifies the "sense" of the flag more -- something like MAY_NEED_FURTHER_NORMALIZATION (but that's too long)

Reply:
It should probably be a new flag that's added to `HAS_PROJECTION`

Reply:
Adding a new flag to `HAS_PROJECTION` then.